src/test/java/io/strimzi/kafka/quotas/TickingClock.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.quotas;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+
+public class TickingClock extends Clock {
+    private Clock baseClock;
+
+    public TickingClock() {
+        this.baseClock = Clock.fixed(Instant.now(), ZoneOffset.UTC);
+    }
+
+    public void tick(Duration tickDuration) {
```

Comment:
Overloading `tick` is a bit awkward with the JDK usage of [tick](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/Clock.html#tick(java.time.Clock,java.time.Duration)). Where it describes `tick` as 
>Obtains a clock that returns instants from the specified clock truncated to the nearest occurrence of the specified duration. 

Which is an off definition to me and this one does what I would expect but maybe we should use a different term?